warehouse/query-core/src/main/java/datawave/next/retrieval/DocumentIteratorOptions.java

```diff
@@ -0,0 +1,220 @@
+            String option = options.get(TYPE_METADATA);
+            try {
+                if (compressedOptions) {
+                    option = decompressOption(option, QueryOptions.UTF8);
+                }
+
+                this.typeMetadata = new TypeMetadata(option);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            throw new RuntimeException("Cannot execute query without TypeMetadata");
+        }
+
+        if (options.containsKey(COMPOSITE_METADATA)) {
+            // no exception is thrown if this key does not exist because CompositeMetadata is not strictly
+            // required. However, evaluation may be adversely affected.
+            String option = options.get(COMPOSITE_METADATA);
+            this.compositeMetadata = CompositeMetadata.fromBytes(java.util.Base64.getDecoder().decode(option));
+        }
+
+        if (options.containsKey(START_TIME) && options.containsKey(END_TIME)) {
+            long start = Long.parseLong(options.get(START_TIME));
+            long end = Long.parseLong(options.get(END_TIME));
+            this.timeFilter = LongRange.of(start, end);
+        } else {
+            throw new RuntimeException("Query must have a time bound");
+        }
+
+        // include fields are optional
+        if (options.containsKey(QueryOptions.PROJECTION_FIELDS)) {
+            String option = options.get(QueryOptions.PROJECTION_FIELDS);
+            // if the user requested everything with a star then leave include fields as null
+            // this signifies that all fields are allowed
+            if (!option.equals("*")) {
+                includeFields = new HashSet<>();
+                includeFields.addAll(Splitter.on(',').splitToList(option));
+            }
+        }
+
+        // technically cannot have both include and exclude fields...but this iterator does not validate optional
+        // options, that is the responsibility of the caller
+        if (options.containsKey(QueryOptions.DISALLOWLISTED_FIELDS)) {
+            String option = options.get(QueryOptions.DISALLOWLISTED_FIELDS);
+            excludeFields = new HashSet<>();
+            excludeFields.addAll(Splitter.on(',').splitToList(option));
+        }
+
+        return true;
+    }
+
+    /**
+     * Direct lift from the QueryIterator
+     *
+     * @param buffer
+     *            the data
+     * @param characterSet
+     *            the character set
+     * @return the decompressed data
+     * @throws IOException
+     *             if there is a deserialization exception
+     */
+    protected String decompressOption(final String buffer, Charset characterSet) throws IOException {
+        final byte[] inBase64 = Base64.decodeBase64(buffer.getBytes());
+
+        ByteArrayInputStream byteInputStream = new ByteArrayInputStream(inBase64);
+        GZIPInputStream gzipInputStream = new GZIPInputStream(byteInputStream);
```

Comment:
Would it be possible to make the compression algorithm configurable or in a follow-on ticket?

Reply:
Absolutely https://github.com/NationalSecurityAgency/datawave/issues/2913